DSharpPlus.Core/Rest/PollyRatelimitPolicy.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+using DSharpPlus.Caching.Abstractions;
+
+using Polly;
+
+namespace DSharpPlus.Core.Rest
+{
+    // main ratelimit handler
+    internal class PollyRatelimitPolicy : AsyncPolicy<HttpResponseMessage>
+    {
+        // keeps track of endpoint -> bucket hash mapping
+        private readonly ConcurrentDictionary<string, string> __endpoint_hash_mapping;
```

Comment:
Dunders belong in python :v 

```suggestion
        private readonly ConcurrentDictionary<string, string> _bucketMap;
```